crates/polars-parquet/src/arrow/write/pages.rs

```diff
@@ -304,12 +308,13 @@ fn expand_list_validity<'a, O: Offset>(
 
         idx += num_zeros;
     }
+    validity.extend_constant(array.values().len() - validity.len(), false);
 
     debug_assert_eq!(idx, array.len());
-
     let validity = validity.freeze();
 
-    array_stack.push_front((array.values().as_ref(), BitmapState::SomeSet(validity)));
+    debug_assert_eq!(validity.len(), array.values().len());
+    array_stack.push((array.values().as_ref(), BitmapState::SomeSet(validity)));
```

Comment:
Instead of extending the bitmap, can't we slice the values with the offsets start/end? Then we serialize the minimal amount of data.

Reply:
I tried this first and it sadly does work. It would require changing significant part of the Parquet writing code.

Most of the code there assumes that it is being handed the whole array and it can slice it up itself.